From: Robert Luberda <robert@debian.org>
Date: Thu, 17 Mar 2011 23:17:51 +0100
Subject: 0027 Include Debian Makefile.languages.inc

This patch adds include ../../debian/local/Makefile.languages.inc
for american/british/english languages and allows to overwrite install
target.

Additionally support for a special `debian-none' value of MASTERHASH
is added into top-level Makefile.
---
 Makefile                    |    2 +-
 languages/american/Makefile |    5 ++++-
 languages/british/Makefile  |    5 ++++-
 languages/english/Makefile  |    6 +++++-
 4 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index b54042f..2503722 100644
--- a/Makefile
+++ b/Makefile
@@ -396,7 +396,7 @@ install-languages:
 	  [ -d $$LIBDIR ]  ||  $(MAKE) NEWDIR=$$LIBDIR mkdirpath; \
 	  set -x; \
 	  cd $$LIBDIR; \
-	  if [ $$MASTERHASH != $$DEFHASH ]; then \
+	  if [ $$MASTERHASH != debian-none ] && [ $$MASTERHASH != $$DEFHASH ]; then \
 	    rm -f $$DEFHASH; \
 	    $$LINK $$MASTERHASH $$DEFHASH; \
 	  fi
diff --git a/languages/american/Makefile b/languages/american/Makefile
index c586109..b3f7013 100644
--- a/languages/american/Makefile
+++ b/languages/american/Makefile
@@ -147,11 +147,14 @@ all:  $(CONFIG)
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+INSTALL_TARGET: install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install:	all $(CONFIG) $(DICTSRC)/english.5
+$(INSTALL_TARGET):	all $(CONFIG) $(DICTSRC)/english.5
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
diff --git a/languages/british/Makefile b/languages/british/Makefile
index 5a961c7..1d4b1bd 100644
--- a/languages/british/Makefile
+++ b/languages/british/Makefile
@@ -147,11 +147,14 @@ all:  $(CONFIG)
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+INSTALL_TARGET := install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install:	all $(CONFIG) $(DICTSRC)/english.5
+$(INSTALL_TARGET):	all $(CONFIG) $(DICTSRC)/english.5
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
diff --git a/languages/english/Makefile b/languages/english/Makefile
index 79062e5..acbb22f 100644
--- a/languages/english/Makefile
+++ b/languages/english/Makefile
@@ -187,11 +187,15 @@ all:  $(CONFIG) english.5
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+
+INSTALL_TARGET := install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install: all $(CONFIG)
+$(INSTALL_TARGET): all $(CONFIG)
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
-- 
